src/Paprika/Store/StorageFanOutPage.cs

```diff
@@ -102,6 +102,30 @@ public void Report(IReporter reporter, IPageResolver resolver, int level)
             }
         }
     }
+    public void Destroy(IBatchContext batch, in NibblePath prefix)
+    {
+        // Destroy the Id entry about it
+        Set(prefix, ReadOnlySpan<byte>.Empty, batch);
```

Comment:
Why call `Set` here? Isn't it handled by the following `DeleteByPrefix`?

Reply:
Destroy should `COW` like `Set` does. The `batchId` check should be done to not write at readonly pages.